### DISCUSSION TITLE
Protobuf schemas that use import

### DISCUSSION THREAD

**Discussion by einarjohnson** [Apicurio/apicurio-registry#1815] [Q&A]
Hi,
I am working on a project that is using Protobuf to define schemas for events that are published to Kafka. We use `import` statements in most messages to re-use definitions. The folder structure is roughly like this

_proto\v1\common
proto\v1\message-domain1\msgtype1.proto, msgtype2.proto, ... + files that are imported by msgtypeN
proto\v1\message-domain2\msgtype1.proto, msgtype2.proto, ... + files that are imported by msgtypeN
..._

where message types are importing either from the common folder or from  proto files within the same directory.

What are the best practices in importing projects like this into the apicurio registry? I am mainly thinking how I can get the imports to work and be recognized by the schema registry. Do I need to flatten out the each _messagetype_ into a single proto file before importing? If so, are there any tools to do this?

**Comment by EricWittmann** [comment]
This is an area of active design/work.  @carlesarnal FYI :)
